src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -521,26 +548,7 @@ StringLiteralEntry *GlobalStringLiteralMap::AddInternedString(STRINGREF *pString
     CONTRACTL_END;
 
     EEStringData StringData = EEStringData((*pString)->GetStringLength(), (*pString)->GetBuffer());
-    StringLiteralEntry *pRet;
-
-    {
-    PinnedHeapHandleBlockHolder pStrObj(&m_PinnedHeapHandleTable,1);
-    SetObjectReference(pStrObj[0], (OBJECTREF) *pString);
-
-    // Since the allocation might have caused a GC we need to re-get the
-    // string data.
-    StringData = EEStringData((*pString)->GetStringLength(), (*pString)->GetBuffer());
-
-    StringLiteralEntryHolder pEntry(StringLiteralEntry::AllocateEntry(&StringData, (STRINGREF*)pStrObj[0]));
-    pStrObj.SuppressRelease();
-
-    // Insert the handle to the string into the hash table.
-    m_StringToEntryHashTable->InsertValue(&StringData, (LPVOID)pEntry, FALSE);
-    pEntry.SuppressRelease();
-    pRet = pEntry;
-    }
-
-    return pRet;
+    return AddStringLiteral(&StringData, preferFrozenObjHeap);
 }
 
 void GlobalStringLiteralMap::RemoveStringLiteralEntry(StringLiteralEntry *pEntry)
```

Comment:
m_PinnedHeapHandleTable.ReleaseHandle later in this method should be executed only if the string is not frozen

Reply:
Thanks, fixed